src/server/tiering/op_manager.h

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <absl/container/flat_hash_map.h>
+
+#include <deque>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "server/tiering/common.h"
+#include "util/fibers/fiber2.h"
+#include "util/fibers/synchronization.h"
+
+namespace dfly::tiering {
+
+// Single-thread MPSC queue for asynchronous storage operations handled by a worker fiber.
```

Comment:
This is nice! So the idea is that each shard get its own `OpQueue` and all tiering operations (indexed) for each key go over that queue. The manager acts like a shard set :)

Reply:
The main idea is that each shard gets its own OpManager, so it can have multiple queues. Having multiple transactions is more difficult

Reply:
I will add a comment about this when I integrate this into the tiered storage